eclipse-collections-code-generator/src/main/resources/impl/iterator/primitiveImmutableSingletonIterator.stg

```diff
@@ -0,0 +1,52 @@
+import "copyright.stg"
+import "primitiveLiteral.stg"
+
+targetPath() ::= "org/eclipse/collections/impl/iterator"
+
+fileName(primitive) ::= "ImmutableSingleton<primitive.name>Iterator"
+
+class(primitive) ::= <<
+<body(primitive.type, primitive.name)>
+>>
+
+body(type, name) ::= <<
+<copyright()>
+
+package org.eclipse.collections.impl.iterator;
+
+import java.util.NoSuchElementException;
+
+import org.eclipse.collections.api.iterator.<name>Iterator;
+
+/**
+ * This file was automatically generated from template file primitiveImmutableSingletonIterator.stg.
+ */
+public final class ImmutableSingleton<name>Iterator implements <name>Iterator
```

Comment:
It's weird to name a class `ImmutableSomething` when the implementation is not immutable. I'd suggest just calling it `Singleton<name>Iterator`. The usage is also not restricted to the immutable classes. For example, this could be used by mutable collections of size 1 as an optimization.